lib/Sema/CSBindings.cpp

```diff
@@ -1629,11 +1667,30 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("delayed");
   if (isSubtypeOfExistentialType())
     attributes.push_back("subtype_of_existential");
-  auto literalKind = getLiteralKind();
+  auto literalKind = getLiteralForScore();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
-    attributes.push_back(literalAttrStr);
+    std::vector<std::string> kinds;
+    std::string literalAttrStr;
+    literalAttrStr.append("[literal: ");
+    if (literalKind == LiteralBindingKind::Atom) {
+      if (auto atomKind = TypeVar->getImpl().getAtomicLiteralKind()) {
+        kinds.push_back(getAtomLiteral(*atomKind).str());
+      }
+    } else if (literalKind == LiteralBindingKind::Collection) {
+      for (const auto &protocolKind : getLiteralRequirements()) {
+        kinds.push_back((getCollectionLiteral(protocolKind) + " " +
+                         getLiteralBindingKind(literalKind))
+                            .str());
+      }
+    } else {
+      kinds.push_back(getLiteralBindingKind(literalKind).str());
+    }
+    llvm::interleave(
```

Comment:
`interleave` in `swift` namespace accepts a range so you could use `interleave(kinds, ...)` instead.